src/main/java/com/simplaex/maven/waitfor/WaitForMojo.java

```diff
@@ -168,15 +170,23 @@ public void execute() throws MojoFailureException {
           }
           try (final CloseableHttpResponse httpResponse = httpClient.execute(httpUriRequest)) {
             final int statusCode = httpResponse.getStatusLine().getStatusCode();
+            final String response = EntityUtils.toString(httpResponse.getEntity());
```

Comment:
I get that you wanted to extract the response out of the if/else below so you can access it later, but now the response is potentiallyconsumed twice. In Line 177 it consumes the response which has always been consumed here already.

You might want to change this to
- always consume the respone or
- consume if response if (chatty or to be checked against an expected response)